### PR TITLE
Twitter

### PR DESCRIPTION
**Pull request by lkurash**

### CODE REVIEW

File: server/error/ApiError.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class ApiError extends Error{
```
Comment (yukas) [inline server/error/ApiError.js]:
```suggestion
class ApiError extends Error {
```

File: client/public/index.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
```
Comment (yukas) [inline client/public/index.html:1]: Remove this file

File: client/src/AppRouter.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        user.setAuth(true);
+      })
+      .finally(() => setLoadingPage(false));
+  }, []);
+
+  if (loadingPage) {
+    return <div>Loading...</div>;
+  }
+  return (
+    <div>
```
Comment (yukas) [inline client/src/AppRouter.js]: Do we need this div?

File: client/src/components/BirthForm.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    "June",
+    "July",
+    "August",
+    "September",
+    "October",
+    "November",
+    "December",
+  ];
+
+  const listDays = (day) => {
```
Comment (yukas) [inline client/src/components/BirthForm.js]: Think of a more compact version of this function

File: client/src/components/BirthForm.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    for (let index = 0; index < 30; index++) {
+      let element = day[index];
+
+      element += 1;
+      day.push(element);
+    }
+    return day;
+  };
+
+  const listYears = (year) => {
```
Comment (yukas) [inline client/src/components/BirthForm.js]: Also more compact version here

File: client/src/components/BirthForm.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+          <option value={month} key={month}>
+            {month}
+          </option>
+        ))}
+      </select>
+
+      <select
+        name="day"
+        onInput={(e) => {
+          {
```
Comment (yukas) [inline client/src/components/BirthForm.js]: Remove the {

File: client/src/components/BirthForm.js [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+            setUserSelectDay(e.target.value);
+          }
+        }}
+        className="signup-birth-form-day"
+      >
+        <option disabled value={0} />
```
Comment (yukas) [inline client/src/components/BirthForm.js]: What is this for?

File: client/src/components/BirthForm.js [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+        {!userSelectMonth && <option disabled>Select month</option>}
+        {userSelectMonth === "February" &&
```
Comment (yukas) [inline client/src/components/BirthForm.js]:
```js
{isFebruary() && listFebruaryDays.map(day => generateDayMarkup(day))}
{is30DaysMonth() && list30Days.map(...)}
{is31DaysMonth() && list31Days.map(...)}
```

File: client/src/components/BirthForm.js [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+        }}
+      >
+        <option disabled />
+        {listYears([1970]).map((year) => (
+          <option value={year} key={year}>
+            {year}
+          </option>
+        ))}
+      </select>
+      {location === SIGNUP_PAGE ? (
```
Comment (yukas) [inline client/src/components/BirthForm.js]: Registration = RegistrationBirthFrom = BirthFrom + his button
ProfileEdit = ProfileEditBirthFrom = BirthFrom + his button

Registration = BirthFrom + his button
ProfileEdit = BirthFrom + his button

**Review by yukas — CHANGES_REQUESTED** [review]

File: client/src/components/BirthForm.js [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+    "January",
+    "March",
+    "May",
+    "July",
+    "August",
+    "October",
+    "December",
+  ];
+
+  const optionDays = [];
```
Comment (yukas) [inline client/src/components/BirthForm.js]: dayOptions

File: client/src/components/BirthForm.js [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+  let birthDate;
+
+  const getYears = () => {
+    for (let year = 1970; year < 2017; year++) {
+      optionYears.push(year);
+    }
+    return optionYears;
+  };
+
+  const getDays = (countDay) => {
```
Comment (yukas) [inline client/src/components/BirthForm.js]: https://stackoverflow.com/questions/3746725/how-to-create-an-array-containing-1-n

File: server/README.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
+# Twitter
+
+# Create Database
```
Comment (yukas) [inline server/README.md]: # Twitter

## Database setup

This project was made with PostgreSQL (https://www.postgresql.org/) and Sequelize (https://sequelize.org/).

### Install npm packages

`npm install`

### Create database config

`cd server`
`cp config/config.example.json config/config.json`

### Create database

`npx sequelize-cli db:create`

### Run migrations

`npx sequelize-cli db:migrate`

### Run seeds

`npx sequelize-cli db:seed:all`

or

`npx sequelize-cli db:seed --seed my-seed`

## Run the app

`npm run dev`

Open [http://localhost:5500](http://localhost:5500) to view it in your browser.

**Review by Steve07101 — APPROVED** [review]

File: server/README.md [hunk 13]
```diff
@@ -0,0 +1,30 @@
+# Twitter
+
+## Database setup
+
+#### This project was made with PostgreSQL (https://www.postgresql.org/) and Sequelize (https://sequelize.org/).
+
```
Comment (yukas) [inline server/README.md:6]: Make `cd server` a first step

File: server/README.md [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+## Run migrations
+### `npx sequelize-cli db:migrate`
+
+## Run seeds
+### `npx sequelize-cli db:seed:all`
+
+### or
+
+### `npx sequelize-cli db:seed --seed my-seed`
```
Comment (yukas) [inline server/README.md]: `npx sequelize-cli db:seed --seed SEED_NAME`

File: client/README.md [hunk 15]
```diff
@@ -0,0 +1,14 @@
+# Getting Started with Create React App
+
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+
+## Available Scripts
+
+In the project directory, you can run:
+
+### `cd client`
+
```
Comment (yukas) [inline client/README.md:10]: You forgot `npm install`

File: client/src/http/userApi.js [hunk 16]
```diff
@@ -0,0 +1,74 @@
+import { $authHost, $host } from ".";
+
+export const authorization = async (email, password) => {
```
Comment (yukas) [inline client/src/http/userApi.js]: Methods named using verbs, not nouns

Comment (yukas) [reply client/src/http/userApi.js]: This is authentication, not an authorization

File: client/src/http/userApi.js [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+
+export const checkToken = async () => {
+  const response = await $authHost.get("api/user/auth");
+
+  localStorage.setItem("token", response.data.token);
+
+  return response.data.token;
+};
+
+export const getUserInfo = async (email) => {
```
Comment (yukas) [inline client/src/http/userApi.js]: "Info" is an empty word. Don't use empty words such as "info", "data", etc.

File: client/src/http/userApi.js [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+  return userInfo.data;
+};
+
+export const updateUserProfile = async (user) => {
+  const userProfile = await $authHost.post("api/user/profile", user);
+
+  return userProfile.data;
+};
+export const getAllUsers = async () => {
+  const users = await $host.get("api/user/allUsers");
```
Comment (yukas) [inline client/src/http/userApi.js]: "api/users"
"api/user/1"

Look at REST notation.

File: client/src/http/userApi.js [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+};
+
+export const getUserInfo = async (email) => {
+  const userInfo = await $authHost.get("api/user/home", email);
+
+  return userInfo.data;
+};
+
+export const updateUserProfile = async (user) => {
+  const userProfile = await $authHost.post("api/user/profile", user);
```
Comment (yukas) [inline client/src/http/userApi.js]: "api/user"

File: client/src/http/userApi.js [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+
+  return users.data;
+};
+export const getUserById = async (id) => {
+  const userById = await $authHost.get(`api/user/${id}`);
+
+  return userById.data;
+};
+export const createFollow = async (followUserId) => {
+  const following = await $authHost.post("api/user/following", {
```
Comment (yukas) [inline client/src/http/userApi.js]: POST: "api/user/{id}/follow" — Look at REST.

File: client/src/http/userApi.js [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+  const users = await $host.get("api/user/allUsers");
+
+  return users.data;
+};
+export const getUserById = async (id) => {
+  const userById = await $authHost.get(`api/user/${id}`);
+
+  return userById.data;
+};
+export const createFollow = async (followUserId) => {
```
Comment (yukas) [inline client/src/http/userApi.js]: `createFollower`

File: client/src/http/userApi.js [hunk 22]
```diff
@@ -0,0 +1,74 @@
...
+  const following = await $authHost.post("api/user/following", {
+    followUserId,
+  });
+
+  return following.data;
+};
+export const deleteFollow = async (followUserId) => {
```
Comment (yukas) [inline client/src/http/userApi.js]: `deleteFollower`

File: client/src/http/userApi.js [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+  const unFollow = await $authHost.post("api/user/deleteFollowing", {
```
Comment (yukas) [inline client/src/http/userApi.js]: Rule: variables are named with nouns: `response`

File: client/src/http/userApi.js [hunk 24]
```diff
@@ -0,0 +1,74 @@
...
+  });
+
+  return unFollow.data;
+};
+export const getFollowingUser = async (id) => {
+  const allFollowing = await $authHost.get(`api/user/following/${id}`);
+
+  return allFollowing.data;
+};
+export const getFollowersUser = async (id) => {
```
Comment (yukas) [inline client/src/http/userApi.js]: `getFollowerUsers`

File: client/src/http/userApi.js [hunk 25]
```diff
@@ -0,0 +1,74 @@
...
+  return following.data;
+};
+export const deleteFollow = async (followUserId) => {
+  const unFollow = await $authHost.post("api/user/deleteFollowing", {
+    followUserId,
+  });
+
+  return unFollow.data;
+};
+export const getFollowingUser = async (id) => {
```
Comment (yukas) [inline client/src/http/userApi.js]: `getFollowingUsers`

File: client/src/http/userApi.js (hunk 23 above)
Comment (yukas) [inline client/src/http/userApi.js]: Make url use the REST notation

File: client/src/http/userApi.js [hunk 26]
```diff
@@ -0,0 +1,74 @@
...
+  const allFollowing = await $authHost.get(`api/user/following/${id}`);
```
Comment (yukas) [inline client/src/http/userApi.js]: Make url use the REST notation

File: client/src/http/userApi.js [hunk 27]
```diff
@@ -0,0 +1,74 @@
...
+
+  return allFollowing.data;
+};
+export const getFollowersUser = async (id) => {
+  const allFollowers = await $authHost.get(`api/user/followers/${id}`);
```
Comment (yukas) [inline client/src/http/userApi.js]: Make url use the REST notation

**Review by yukas — CHANGES_REQUESTED** [review]

File: client/src/router.js [hunk 28]
```diff
@@ -0,0 +1,143 @@
...
+import MessagesPageComponent from "./authPages/MessagePageComponent";
+import WrapperTwitterPage from "./pages/WrapperTwitterPage";
+import authorization from "./utils/authorization";
+import ProfilePageLikes from "./components/ProfilePageLikes";
+import ProfilePageMedia from "./components/ProfilePageMedia";
+import ProfilePageAnswers from "./components/ProfilePageAnswers";
+import UserTwits from "./components/UserTwits";
+import WrapperLoginAndSignUpPage from "./pages/WrapperLoginAndSignUpPage";
+
+export const authRoutes = {
```
Comment (yukas) [inline client/src/router.js:41]: `authenticatedRoutes` or `privateRoutes`

File: client/src/router.js [hunk 29]
```diff
@@ -0,0 +1,143 @@
...
+  path: HOME_PAGE,
+  element: <WrapperTwitterPage />,
```
Comment (yukas) [inline client/src/router.js]: This component should be named `Layout` or `AppLayout` etc

File: client/src/router.js [hunk 30]
```diff
@@ -0,0 +1,143 @@
...
+  loader: authorization,
```
Comment (yukas) [inline client/src/router.js]: Better name would be `authenticate`

File: client/src/router.js [hunk 31]
```diff
@@ -0,0 +1,143 @@
...
+import WrapperTwitterPage from "./pages/WrapperTwitterPage";
+import authorization from "./utils/authorization";
+import ProfilePageLikes from "./components/ProfilePageLikes";
+import ProfilePageMedia from "./components/ProfilePageMedia";
+import ProfilePageAnswers from "./components/ProfilePageAnswers";
+import UserTwits from "./components/UserTwits";
+import WrapperLoginAndSignUpPage from "./pages/WrapperLoginAndSignUpPage";
+
+export const authRoutes = {
+  path: HOME_PAGE,
```
Comment (yukas) [inline client/src/router.js]: `HOME_PAGE_PATH`

File: client/src/utils/constans.js [hunk 32]
```diff
@@ -0,0 +1,24 @@
...
+
+export const TWITTER_USER_PAGE = "/twitter/profile/";
+
+export const PROFILE_PAGE_USER = "/auth/profile/";
+export const PROFILE_PAGE_USER_LIKES = "/auth/profile/:id/likes";
+export const PROFILE_PAGE_USER_TWITS = "/auth/profile/:id/twits";
+export const PROFILE_PAGE_USER_ANSWERS = "/auth/profile/:id/answers";
+export const PROFILE_PAGE_USER_MEDIA = "/auth/profile/:id/media";
+
+export const BOOKMARKS_PAGE = "/auth/bookmarks/";
```
Comment (yukas) [inline client/src/utils/constans.js]: Move :id segment inside the contant

File: client/src/router.js [hunk 33]
```diff
@@ -0,0 +1,143 @@
...
+    },
+    {
+      path: `${AUTHEXPLORE_PAGE}`,
+      element: <ExplorePage />,
+    },
+  ],
+};
+
+export const publicRoutes = {
+  path: TWITTER_PAGE,
```
Comment (yukas) [inline client/src/router.js]: `ROOT_PAGE`

File: client/src/router.js [hunk 34]
```diff
@@ -0,0 +1,143 @@
...
+  element: <WrapperTwitterPage />,
```
Comment (yukas) [inline client/src/router.js]: `PublicLayout`

File: client/src/router.js [hunk 35]
```diff
@@ -0,0 +1,143 @@
...
+  children: [
+    {
+      path: "",
+      element: <TwitterPage />,
```
Comment (yukas) [inline client/src/router.js]: `PublicHomePage`

File: client/src/router.js [hunk 36]
```diff
@@ -0,0 +1,143 @@
...
+      element: <ExplorePage />,
+    },
+    {
+      path: `${TRENDS_PAGE}:trend`,
+      element: <TrendsPage />,
+    },
+    {
+      path: `${TWITTER_USER_PAGE}:id`,
+      element: <TwitterPageNotAuthUser />,
+    },
```
Comment (yukas) [inline client/src/router.js]: Login and signup should be inside public routes, no need for a separate grouping

File: client/src/components/forms/LoginForm.js [hunk 37]
```diff
@@ -0,0 +1,82 @@
...
+import LoginEmailForm from "./LoginEmailForm";
+import Logo from "../common/Logo";
+
+import "./loginAndRegistretionForm.css";
+
+const LoginForm = observer(() => {
+  const { usersStore } = useContext(Context);
+  const navigate = useNavigate();
+  const [showFormPassword, setShowFormPassword] = useState(false);
+  const [showFormEmail, setShowFormEmail] = useState(true);
```
Comment (yukas) [inline client/src/components/forms/LoginForm.js]:
```suggestion
  const [passwordFieldVisible, setPasswordFieldVisible] = useState(false);
  const [emailFieldVisible, setemailFieldVisible] = useState(true);
```

File: client/src/components/forms/LoginForm.js [hunk 38]
```diff
@@ -0,0 +1,82 @@
...
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const location = useLocation().pathname;
+
+  const showPasswordForm = () => {
```
Comment (yukas) [inline client/src/components/forms/LoginForm.js]:
```suggestion
  const showPasswordField = () => {
```

File: client/src/components/forms/LoginForm.js [hunk 39]
```diff
@@ -0,0 +1,82 @@
...
+    }
+  };
+
+  const signIn = async () => {
+    try {
+      if (email && password) {
+        const userProfile = await authentication(email, password);
+
+        usersStore.setUser(userProfile);
+        usersStore.setAuth(true);
```
Comment (yukas) [inline client/src/components/forms/LoginForm.js]: Try to follow the "Single level of abstraction rule" here:

```suggestion
        const authenticationResult = await authClient.authenticate(email, password);

        usersStore.setUserAuth(authenticationResult.user);
        localAuthClient.setAuthToken(authenticationResult.token);
        
```

File: client/src/components/forms/LoginForm.js [hunk 40]
```diff
@@ -0,0 +1,82 @@
...
+        showFormEmail={showFormEmail}
+        setShowFormEmail={setShowFormEmail}
+      />
+
+      <LoginPasswordForm
+        email={email}
+        password={password}
+        showFormPassword={showFormPassword}
+        setPassword={setPassword}
+        signInKeyDown={signInKeyDown}
```
Comment (yukas) [inline client/src/components/forms/LoginForm.js]: This seems should be inside the component itself.

File: client/src/components/forms/LoginForm.js [hunk 41]
```diff
@@ -0,0 +1,82 @@
...
+        signIn={signIn}
+      />
+    </>
+    // <div className="body">
```
Comment (yukas) [inline client/src/components/forms/LoginForm.js]:
```suggestion
      if (showPasswordForm) {
        <LoginEmailForm
          email={email}
          setEmail={setEmail}
          showFormEmail={showFormEmail}
          setShowFormEmail={setShowFormEmail}
        />
      } else {
        <LoginPasswordForm
```

File: client/src/router.js [hunk 42]
```diff
@@ -0,0 +1,154 @@
+import AppLayout from "./pages/AppLayout";
```
Comment (yukas) [inline client/src/router.js]: Imports structure:
```
// utils
// public
// private
// components

or
// utils
// pages
// components
```

File: client/src/router.js [hunk 43]
```diff
@@ -0,0 +1,154 @@
...
+  AUTHTRENDS_PAGE_PATH,
+} from "./utils/constans";
+import redirectLoader from "./utils/redirectLoader";
+
+export const publicRoutes = {
+  path: ROOT_PAGE_PATH,
+  element: <AppLayout />,
+  children: [
+    {
+      path: "",
```
Comment (yukas) [inline client/src/router.js]: `EXPLORE_PATH_ROOT` or some other constant, but not the string itself

File: client/src/router.js [hunk 44]
```diff
@@ -0,0 +1,154 @@
...
+      element: <PublicHomePage />,
+      loader: redirectLoader,
+    },
+    {
+      path: EXPLORE_PAGE_PATH,
+      element: <ExplorePage />,
+    },
+    {
+      path: `${TRENDS_PAGE_PATH}:trend`,
```
Comment (yukas) [inline client/src/router.js]: Add `:trend` part into the constant

File: client/src/router.js [hunk 45]
```diff
@@ -0,0 +1,154 @@
...
+    },
+    {
+      path: `${TWITTER_USER_PAGE_PATH}:id`,
+      element: <PublicProfilePageUser />,
+    },
+  ],
+};
+
+export const authorizationRoutes = {
+  path: "/authentication",
```
Comment (yukas) [inline client/src/router.js]: `auth` is a better name here

File: client/src/router.js [hunk 46]
```diff
@@ -0,0 +1,154 @@
...
+  element: <LayoutLoginAndSignUpPage />,
+  loader: redirectLoader,
+  children: [
+    {
+      path: "/authentication/redirect",
```
Comment (yukas) [inline client/src/router.js]: `auth/login/redirect`

File: client/src/router.js [hunk 47]
```diff
@@ -0,0 +1,154 @@
...
+  ],
+};
+
+export const privateRoutes = {
+  path: HOME_PAGE_PATH,
+  element: <AppLayout />,
+  loader: authenticate,
+  children: [
+    {
+      path: "",
```
Comment (yukas) [inline client/src/router.js]: Use constant here

File: client/src/router.js [hunk 48]
```diff
@@ -0,0 +1,154 @@
...
+      element: <HomePageComponent />,
+    },
+    {
+      path: `${AUTHEXPLORE_PAGE_PATH}`,
```
Comment (yukas) [inline client/src/router.js]: No need for string interpolation here

File: client/src/router.js [hunk 49]
```diff
@@ -0,0 +1,154 @@
...
+      element: <ExplorePage />,
+    },
+    {
+      path: `${NOTIFICATIONS_PAGE_PATH}`,
+      // element: <UserPageComponent />,
```
Comment (yukas) [inline client/src/router.js]: Don't comment out things, use git to save things for future if needed.

File: client/src/router.js [hunk 50]
```diff
@@ -0,0 +1,154 @@
...
+          path: `${PROFILE_PAGE_USER_LIKES_PATH}`,
+          element: <ProfilePageLikes />,
+        },
+        {
+          path: `${PROFILE_PAGE_USER_MEDIA_PATH}`,
+          element: <ProfilePageMedia />,
+        },
+        {
+          path: `${EDIT_PROFILE_PAGE_PATH}`,
+          element: [<UserTwits key={'twits'}/>, <EditProfilePageComponent key={'form'}/>],
```
Comment (yukas) [inline client/src/router.js]: Create a separate component for this

File: client/src/utils/constans.js [hunk 51]
```diff
@@ -0,0 +1,45 @@
+export const ROOT_PAGE_PATH = "/";
+
+export const PUBLIC_HOME_PAGE_PATH = "";
+export const EXPLORE_PAGE_PATH = "/explore";
+export const PUBLIC_USERS_PAGE_PATH = "/profile/";
```
Comment (yukas) [inline client/src/utils/constans.js]: Use `/user/` instead of `/profile/`

File: client/src/utils/constans.js [hunk 52]
```diff
@@ -0,0 +1,45 @@
...
+export const AUTHTREND_PAGE_PATH = "/home/trends/:trend";
+
+export const PROFILE_PAGE_USER_PATH = "/home/profile";
+export const PROFILE_PAGE_USER_TWITS_PATH = "";
+export const PROFILE_PAGE_USER_LIKES_PATH = "/home/profile/likes";
+export const PROFILE_PAGE_USER_ANSWERS_PATH = "/home/profile/answers";
+export const PROFILE_PAGE_USER_MEDIA_PATH = "/home/profile/media";
+export const EDIT_PROFILE_PAGE_PATH = "/home/profile/edit";
+
+export const PRIVATE_USER_PAGE_PATH = "/:id";
```
Comment (yukas) [inline client/src/utils/constans.js]: The "private" prefix seems pretty useless here, try to remove it

File: client/src/utils/constans.js [hunk 53]
```diff
@@ -0,0 +1,45 @@
+export const ROOT_PAGE_PATH = "/";
+
+export const PUBLIC_HOME_PAGE_PATH = "";
+export const EXPLORE_PAGE_PATH = "/explore";
+export const PUBLIC_USERS_PAGE_PATH = "/profile/";
+export const PUBLIC_USER_PAGE_PATH = "/profile/:id";
+export const TRENDS_PAGE_PATH = `/trends/`;
+export const TREND_PAGE_PATH = `/trends/:trend`;
+
```
Comment (yukas) [inline client/src/utils/constans.js:7]: Add a comment to the each group

File: client/src/router.js [hunk 54]
```diff
@@ -0,0 +1,206 @@
...
+      element: <TrendsPage />,
+    },
+    {
+      path: PUBLIC_USER_PAGE_PATH,
+      element: <PublicProfilePageUser />,
+    },
+  ],
+};
+
+export const authorizationRoutes = {
```
Comment (yukas) [inline client/src/router.js]: `authorizationRoutes`

File: client/src/http/index.js [hunk 55]
```diff
@@ -0,0 +1,21 @@
+import axios from "axios";
```
Comment (yukas) [inline client/src/http/index.js]: Rename this file to outline what really done in it

File: client/src/http/index.js [hunk 56]
```diff
@@ -0,0 +1,21 @@
...
+const Cookies = require("js-cookie");
+
+const $host = axios.create({
+  baseURL: "http://localhost:5500",
```
Comment (yukas) [inline client/src/http/index.js]: Move it to the constant

File: client/src/http/userApi.js [hunk 57]
```diff
@@ -0,0 +1,91 @@
+import { $authHost, $host } from ".";
+
+class UserApi {
```
Comment (yukas) [inline client/src/http/userApi.js]: The class should be named `UserClient`

File: client/src/http/twitsApi.js [hunk 58]
```diff
@@ -0,0 +1,139 @@
+import { $authHost, $host } from ".";
+
+class TwitsApi {
```
Comment (yukas) [inline client/src/http/twitsApi.js]: `TwitsClient`

File: client/src/http/trendsApi.js [hunk 59]
```diff
@@ -0,0 +1,42 @@
+import { $host } from ".";
+
+class Trends {
```
Comment (yukas) [inline client/src/http/trendsApi.js]: `TrendsClient`

File: client/src/http/trendsApi.js [hunk 60]
```diff
@@ -0,0 +1,42 @@
...
+    return trend.data;
+  }
+
+  async getCountTrends(text) {
+    const trends = await $host.put("api/twitter/trends/countTrends", text);
+
+    return trends.data;
+  }
+}
+const trendsApi = new Trends();
```
Comment (yukas) [inline client/src/http/trendsApi.js]: `const trendsClient = ...`

File: client/src/http/trendsApi.js [hunk 61]
```diff
@@ -0,0 +1,42 @@
...
+    const twits = await $host.get(`api/twitter/trends/${trend}`, {
+      params: { limit, list },
+    });
+
+    return twits.data;
+  }
+
+  async createNotInterestingTrend(trendId, userId) {
+    const trend = await $host.put(
+      `api/twitter//trends/trend/${trendId}/user/${userId}`
```
Comment (yukas) [inline client/src/http/trendsApi.js]: Remove unneded slash

**Pull request merged**